osidb/models.py

```diff
@@ -1050,6 +1055,43 @@ def _validate_ps_module_new_flaw(self):
                     f"for flaw with bz_id {bz_id}."
                 )
 
+    def _validate_sofware_collection(self):
+        """
+        Check that all RHSCL components in flaw's affects start with a valid collection.
+        """
+        if (
+            not self.ps_module.startswith("rhscl")
+            or self.ps_component in COMPONENTS_WITHOUT_COLLECTION
+        ):
+            return
+
+        streams = PsUpdateStream.objects.filter(ps_module__name=self.ps_module)
+        collections = streams.values_list("collections", flat=True).all()
+
+        is_valid_component = False
+        is_meta_package = False
+        for collection in collections:
+            for component in collection:
+                if self.ps_component.startswith(component + "-"):
+                    is_valid_component = True
+                    break
+                elif self.ps_component == component:
+                    is_meta_package = True
+                    break
+
+        if is_meta_package:
+            self.alert(
+                "flaw_affects_rhscl_collection_only",
+                f"PSComponent {self.ps_component} for {self.ps_module} indicates collection "
+                "meta-package rather than a specific component in the collection",
+            )
+        elif not is_valid_component:
+            self.alert(
+                "flaw_affects_rhscl_invalid_collection",
+                f"PSComponent {self.ps_component} for {self.ps_module} "
+                "does not match any valid collection",
+            )
```

Comment:
Well this still doesn't allow for both alerts to occur at the same time since there is the `if .. elif` statement. To completely mimic the SFM2 behavior you need to have it as a separate `if` statements as in:
https://git.prodsec.redhat.com/devops/sfm2/-/blob/master/sfm2/validation/flaw.py#L1996
https://git.prodsec.redhat.com/devops/sfm2/-/blob/master/sfm2/validation/flaw.py#L2010

Reply:
Sorry I believed it would not be possible to both occurs at the same time because one validate exact `ps_component == valid_component` and the other compares `self.ps_component.startswith(valid_component + "-")` but having a list of valid components they may have both warnings, I will refactor.

Reply:
I have refactored the validation and the tests to reflect the changes.